tests/v2/validation/provisioning/rke2/etcd_bkp_restore_test.go

```diff
@@ -0,0 +1,515 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/api/scheme"
+	kubeProvisioning "github.com/rancher/rancher/tests/framework/clients/provisioning"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	provisioningV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/ingresses"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/deployments"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultNamespace = "default"
+)
+
+type RKE2EtcdSnapshotRestoreTestSuite struct {
+	suite.Suite
+	session            *session.Session
+	client             *rancher.Client
+	clusterName        string
+	ns                 string
+	kubernetesVersions []string
+	cnis               []string
+	providers          []string
+	nodesAndRoles      []machinepools.NodeRoles
+}
+
+func (p *RKE2EtcdSnapshotRestoreTestSuite) TearDownSuite() {
+	p.session.Cleanup()
+}
+
+var EtcdSnapshotGroupVersionResource = schema.GroupVersionResource{
+	Group:    "rke.cattle.io",
+	Version:  "v1",
+	Resource: "etcdsnapshots",
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	r.kubernetesVersions = clustersConfig.KubernetesVersions
+	r.cnis = clustersConfig.CNIs
+	r.providers = clustersConfig.Providers
+	r.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clusterName = r.client.RancherConfig.ClusterName
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) createSnapshot(clustername string, generation int) error {
+	kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
```

Comment:
This is the wrong client for our validation tests. We need to use the Steve client for this. Please rebase your branch to get the latest updates.

Reply:
This client is for bypassing the Steve v1 API to go through the underlying Kubernetes API, which we don't want to do for validation tests, since we want to test v1 calls through Steve. Make sense?

Reply:
Made the changes so that Steve client is used for v1 calls.